File: YoutubeRadioApplication/src/main/java/project/controller/MusicController.java
package project.controller;

import music.manager.model.UserMusicStatusDTO;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
import project.service.music.inter.IMusicService;

import javax.ws.rs.QueryParam;
import java.net.URI;

/**
 * Created by ldalzotto on 03/05/2017.
 */
@Controller
public class MusicController {

    private static final Logger LOGGER = LoggerFactory.getLogger(MusicController.class.getName());

    @Autowired
    IMusicService iMusicService;

    @RequestMapping(value = "user/{username}/music/musicplatform/{musicplatform}", method = RequestMethod.POST)
    public ResponseEntity<?> addMusicFromRessourceUrl(@PathVariable("username") String username,
                                                      @PathVariable("musicplatform") String musicPlatform, @QueryParam("url") String url) {
        iMusicService.addMusicFromUrlAndMusicPlatform(username, url, musicPlatform);
        URI location = ServletUriComponentsBuilder.fromCurrentRequestUri().build().toUri();
        return ResponseEntity.created(location).build();
    }

    @RequestMapping(value = "user/{username}/music/all", method = RequestMethod.GET)
    public ResponseEntity<UserMusicStatusDTO> getAllMusicUrls(@PathVariable("username") String username) {
        return ResponseEntity.ok(iMusicService.getMusicStatusOfCurrentClient(username));
    }


}

File: YoutubeRadioApplication/src/scala/com/ldz/project/controller/MusicController.scala
package com.ldz.project.controller

import java.net.URI
import javax.ws.rs.QueryParam

import com.ldz.converter.container.ConverterContainer
import com.ldz.project.model.RawMusicStatus
import com.ldz.project.service.music.IMusicService
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.{PathVariable, RequestMapping, RequestMethod}
import org.springframework.web.servlet.support.ServletUriComponentsBuilder
import collection.JavaConverters._


/**
  * Created by ldalzotto on 24/06/2017.
  */
@Controller
class MusicController {
  private val LOGGER = LoggerFactory.getLogger(classOf[MusicController].getName)

  @Autowired
  private val iMusicService: IMusicService = null

  @Autowired
  private val converterContainer: ConverterContainer = null

  @RequestMapping(value = Array("user/{username}/music/musicplatform/{musicplatform}"), method = Array(RequestMethod.POST))
  def addMusicFromRessourceUrl(@PathVariable("username") username: String, @PathVariable("musicplatform") musicPlatform: String, @QueryParam("url") url: String): ResponseEntity[_] = {
    iMusicService.addMusicFromUrlAndMusicPlatform(username, url, musicPlatform) match {
      case true =>
        val location = ServletUriComponentsBuilder.fromCurrentRequestUri.build.toUri
        ResponseEntity.created(location).build()
      case false =>
        ResponseEntity.unprocessableEntity().build()
    }
  }

  @RequestMapping(value = Array("user/{username}/music/all"), method = Array(RequestMethod.GET))
  def getAllMusicUrls(@PathVariable("username") username: String): ResponseEntity[RawMusicStatus] = {
    ResponseEntity.ok(converterContainer.convert(iMusicService.getMusicStatusOfCurrentClient(username), classOf[RawMusicStatus]))
  }

  @RequestMapping(value = Array("user/{username}/music/artists/all-distinct"), method = Array(RequestMethod.GET))
  def getAllDistictMusicArtists(@PathVariable("username") username: String): ResponseEntity[java.util.List[String]] = {
      ResponseEntity.ok(iMusicService.getAllDistinctArtistsFromClient(username).asJava)
  }

  @RequestMapping(value = Array("user/{username}/music/artists/search/{artist-input}"), method = Array(RequestMethod.GET))
  def researchArtistFromInput(@PathVariable("username") username: String, @PathVariable("artist-input") artistInput: String): ResponseEntity[java.util.List[String]] = {
    ResponseEntity.ok(iMusicService.getFilteredArtistsFromInput(username, artistInput).asJava)
  }

}
